src/main/java/io/vertx/httpproxy/interceptors/impl/HeadersInterceptorImpl.java

```diff
@@ -19,29 +19,30 @@
 import io.vertx.httpproxy.ProxyRequest;
 import io.vertx.httpproxy.ProxyResponse;
 
+import java.util.Objects;
 import java.util.Set;
 
 /**
  * The general interceptor for headers. Extended by other implementations.
  */
 public class HeadersInterceptorImpl implements ProxyInterceptor {
-  Handler<MultiMap> changeRequestHeaders;
-  Handler<MultiMap> changeResponseHeaders;
+  private final Handler<MultiMap> changeRequestHeaders;
+  private final Handler<MultiMap> changeResponseHeaders;
 
   public HeadersInterceptorImpl(Handler<MultiMap> changeRequestHeaders, Handler<MultiMap> changeResponseHeaders) {
+    Objects.requireNonNull(changeRequestHeaders);
+    Objects.requireNonNull(changeResponseHeaders);
```

Comment:
Not mandatory but, to reduce the number of lines, you can assign directly from the result of `requireNonNull`.

Like `this.changeRequestHeaders = Objects.requireNonNull(changeRequestHeaders)`